### DISCUSSION TITLE
ShortlinkZoneSSLCertEast: Failed to create resource. Certificate not issued in time

### DISCUSSION THREAD

**Discussion by esmo** [mozilla/hubs#3954] [Hubs Cloud Questions] [answered]
Hey there,

I just tried to create a Hubs stack on AWS and I'm stuck. When I try to deploy the stack I get an error with the message mentioned in the title, which causes a rollback.

I registrered two domains with Route 53 yesterday. Both seem to be set up correctly.

I used recipe 3 for the domain setup, since we want to use a subdomain which is hosted elsewhere. But this should not affect the certificate for the short link domain, as far as I understand.

Any hint is very much appreciated.

**Comment by esmo** [comment, marked as answer]
Just for the record: I registered a new short link domain (the last one was not short at all), and now the deployment worked flawlessly. Maybe the length of the domain name had some side effect? 

Anyways, the stack just completed, and I'm curious to get started.

**Comment by blairmacintyre** [comment]
I don't think the length matters.  Is there any chance the DNS was misconfigured, either because you moved the domain from a different AWS account or from another registrar (I had this problem).

**Reply by josedc** [reply]
Hi @blairmacintyre ... how did you solve the problem after moving the domain from another AWS account?

I spent most of yesterday trying to migrate Hubs Cloud Personal from my personal AWS account in Oregon (working ok since last year) to the school AWS account that I also manage but encountered some error also causing rollbacks: 
1) I was able to move 2 domains (kglam.life and kglam.link) and created 3rd (kg360.link) to the school account, 
2) I tried running the Hubs CloudFormation stack in 
  a) Oregon, 
  b) N Virginia and 
  c) Japan but all failed, 
3) I've tried both Hubs DNS recipes 1 and 2 but no success.

Looks like the same ShortlinkZoneSSLCertEast error reported by @esmo:
![Screenshot 2021-05-17 091838](https://user-images.githubusercontent.com/171154/118430005-a243bd80-b705-11eb-9801-e19e0dcc5437.jpg)

After the last rollback from N Virginia, my recently purchased kg360.link (specified as the internal private domain) looks like this:
![Screenshot 2021-05-17 114354](https://user-images.githubusercontent.com/171154/118430159-06668180-b706-11eb-8433-16fabb39a0aa.jpg)

My kglam.life (specified as the main Hubs domain in DNS recipe 1 or hubs.kglam.life in DNS recipe 2) looks like this:
![Screenshot 2021-05-17 114226](https://user-images.githubusercontent.com/171154/118430238-40d01e80-b706-11eb-83db-0058c0207a7c.jpg)

My kglam.link (specified as the short link domain in DNS recipe 1 and 2) looks like this:
![Screenshot 2021-05-17 114247](https://user-images.githubusercontent.com/171154/118430286-5ba29300-b706-11eb-9db7-3be364d02c2f.jpg)

What is preventing the issue of certificates (from the error message) in time?

Regards,
Jose

**Reply by josedc** [reply]
On more investigation, I realized the Name Servers in "Route53 > Domains > Registered domains" was not the same as the Name Servers in "Route53 > Hosted Zones" for the transferred Hubs and short link domains. I think this was preventing the certificates from being created. However, after I fixed this by making the name servers the same the deleted and re-launched the CloudFormation stack, I got this error:
![Screenshot 2021-05-17 160229](https://user-images.githubusercontent.com/171154/118457104-0a0cff00-b72c-11eb-9266-108a02c2cd2d.jpg)

Anybody have a clue what causes this error?

Resource handler returned message: "Invalid request provided: One or more of the CNAMEs you provided are already associated with a different resource. (Service: CloudFront, Status Code: 409, Request ID: e236765f-8cf7-4ba2-ad6e-350a9a4229e5, Extended Request ID: null)" (RequestToken: c3856874-c82b-89d2-cb00-1c373be11b4b, HandlerErrorCode: InvalidRequest)

I made sure to delete all unnecessary records from the Hosted Zones leaving only the bare NS and SOA records so I don't understand why the CloudFront complains about CNAME's already associated with different resource.

**Reply by esmo** [reply]
@josedc It's now been a while since I fiddled around with this, so I don't really know. I vaguely remember that I had an issue with the main domain, which was hosted with another domain provider. Maybe you have a similar setup, and it's about the CNAME entry you made there, not in Route53?

**Reply by josedc** [reply]
Thanks for the hint @esmo .. I found and fixed the problem. Although the domain names and CNAME entries in the new AWS account were clean, somehow there were unused CloudFront entries in the old AWS account still affecting the new CloudFormation stack. It's strange there is no way to see these from the new AWS account and the only way is to search for them in the old AWS account. After deleting these zombie CloudFront entries the new Hubs stack spun up without any errors.
